packages/core/src/browser/messaging/ws-connection-provider.ts

```diff
@@ -47,10 +47,11 @@ export class WebSocketConnectionProvider {
         const socket = this.createWebSocket(url);
         socket.onerror = console.error;
         socket.onclose = ({ code, reason }) => {
-            for (const channel of this.channels.values()) {
+            const channels = [...this.channels.values()];
+            this.channels.clear();
```

Comment:
You create a list of channels to then clear the list? I am confused.

Reply:
It is not the same list, my bad.

Reply:
`fireClose` can cause modifications to `this.channels` inside the loop, like adding new channels. in order to avoid it, I copy `this.channels` first